component/objectstorage.jsonnet

```diff
@@ -241,9 +242,124 @@ local compositionExoscale =
+          {
+            apiVersion: 'v1',
+            kind: 'Service',
+            name: 'minio-server',
+            fieldPath: 'spec.clusterIP',
+            toConnectionSecretKey: 'ENDPOINT_URL',
+            namespace: 'minio',
+          },
+          {
+            apiVersion: 'v1',
+            kind: 'Secret',
+            name: 'minio-server',
+            fieldPath: 'data.rootUser',
+            toConnectionSecretKey: 'AWS_ACCESS_KEY_ID',
+            namespace: 'minio',
+          },
+          {
+            apiVersion: 'v1',
+            kind: 'Secret',
+            name: 'minio-server',
+            fieldPath: 'data.rootPassword',
+            toConnectionSecretKey: 'AWS_SECRET_ACCESS_KEY',
+            namespace: 'minio',
+          },
+        ],
+        writeConnectionSecretToRef: {
+          name: '',
+          namespace: 'syn-crossplane',
+        },
+        forProvider+: {
+          namespace: 'minio',
+          chart: {
+            name: 'minio',
+            repository: 'https://charts.min.io/',
+            version: '5.0.7',
+          },
+          set: [
+            {
+              name: 'rootUser',
+              value: 'minioadmin',
+            },
+            {
+              name: 'rootPassword',
+              value: 'minioadmin',
+            },
+          ],
+          values: {
+            fullnameOverride: 'minio-server',
+            replicas: 1,
+            resources: {
+              requests: {
+                memory: '128Mi',
+              },
+            },
+            persistence: {
+              size: '1Gi',
+            },
+            mode: 'standalone',
+            buckets: [
+              {
+                name: '',
+                policy: 'none',
+              },
+            ],
+          },
+        },
+        providerConfigRef: {
+          name: 'helm',
+        },
+      },
+    };
+
+  kube._Object('apiextensions.crossplane.io/v1', 'Composition', 'dev.objectbuckets.appcat.vshn.io') +
+  common.SyncOptions +
+  {
+    spec: {
+      compositeTypeRef: comp.CompositeRef(xrd),
+      writeConnectionSecretsToNamespace: 'syn-crossplane',
+      resources: [
+        {
+          base: namespace,
+          patches: [
+            comp.FromCompositeFieldPath('metadata.labels[crossplane.io/composite]', 'spec.forProvider.manifest.metadata.name'),
+          ],
+        },
+        {
+          base: devMinioHelmChart,
+          connectionDetails: [
+            { fromConnectionSecretKey: 'AWS_SECRET_ACCESS_KEY' },
+            { fromConnectionSecretKey: 'ENDPOINT_URL' },
+            { fromConnectionSecretKey: 'AWS_ACCESS_KEY_ID' },
+          ],
+          patches: [
+            comp.FromCompositeFieldPath('metadata.labels[crossplane.io/composite]', 'spec.forProvider.values.buckets[0].name'),
+            comp.FromCompositeFieldPath('metadata.labels[crossplane.io/composite]', 'spec.writeConnectionSecretToRef.name'),
+          ],
+        },
+      ],
+    },
+  };
+
```

Comment:
I'd rather not mix "test-code" and actual code. I'd suggest to move this to `tests/component/objectstorage.jsonnet` or something like that.